### DISCUSSION TITLE
MQTT with client certificate

### DISCUSSION THREAD

**Discussion by j-hallier** [tbnobody/OpenDTU#780] [Q&A] [answered]
Hi,
I want to connect my OpenDTU device to a local mosquitto server which is configured to use TLS and client certificate authentication ("require_certificate true"). So clients are not possible to connect with their username/password combination but have to provide a client certificate and key.
I've tried to misuse user and password entries in the web interface for the certificates but did not succeed. Is it generally possible to add certificates to the OpenDTU device and is the mqtt-part of the SW able to handle client certification at all ?

**Comment by madmartin** [comment]
Hello @j-hallier ,
TLS client authentication for MQTT  is currently not implemented in openDTU.
You can either file an "Request for feature" issue or add this feature and file a pull request.
My personal opionion is that it may be problematic to implement this, because it will use quite a lot of resources in a limited microcontroller environment.

**Comment by madmartin** [comment]
Hello @j-hallier ,
I am preparing a pull request for SSL/TLS cert based login. Should be ready the next days.

**Reply by j-hallier** [reply]
Hi MadMartin, thank you very much. It's working very well. Damn good job !!!

**Comment by madmartin** [comment, marked as answer]
@j-hallier  see #801  - please mark this question "answered" when this fits your needs.
